handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1517,7 +1520,7 @@ public final synchronized boolean isOutboundDone() {
                 for (int i = 0; i < enabled.length; i++) {
                     String mapped = toJavaCipherSuite(enabled[i]);
                     final String cipher = mapped == null ? enabled[i] : mapped;
-                    if (!OpenSsl.isTlsv13Supported() && SslUtils.isTLSv13Cipher(cipher)) {
+                    if ((!tls13Enabled || !OpenSsl.isTlsv13Supported()) && SslUtils.isTLSv13Cipher(cipher)) {
```

Comment:
Does this just mask the problem? My concern is that the contents of `String[] enabled` are what are truly enabled by the SSL engine and this is just filtering them out.

Reply:
Nope ... tls 1.3 ciphers are not "usuable" when tls1.3 is not enabled

Reply:
I believe that. My concern is this, and I'm trying to write a test for it but having issues with the build so bear with me:
```
serverEngine = getEngine(Tlsv3Enabled)
serverEngine.setEnabledCipherSuites(new String[]{ "A_SINGLE_TLS1.2_CIPHER" })
```
In my minds eye, that is implicitly disabling tlsv3, but I'm not confident that is really happening and it will end up supporting the v1.3 cipher suites.

Reply:
This test demonstrates what I'm afraid of: https://github.com/bryce-anderson/netty/commit/294cc77446d9eb86423b96455e7b680826053f18

Reply:
Good catch... there was even a TODO :D PTAL again